src/main/java/de/dennisguse/opentracks/fragments/StatisticsRecordedFragment.java

```diff
@@ -161,13 +161,25 @@ public void loadStatistics() {
                         sharedPreferenceChangeListener.onSharedPreferenceChanged(sharedPreferences, getString(R.string.stats_rate_key));
                     }
 
+                    loadTrackDescription(track);
                     updateUI();
                     updateSensorUI();
                 }
             });
         }
     }
 
+    private void loadTrackDescription(Track track) {
+        if (track == null) {
```

Comment:
why can the track be null?

Reply:
I don't know either :smile: 

But, look at 155 and 158 lines.

```
155: trackStatistics = track != null ? track.getTrackStatistics() : null;
...
158: String newCategory = track != null ? track.getCategory() : "";

```

Reply:
Well then for now ;)
All these null checks are kind of useless as the activity should abort onCreate if no track is there, right?

Reply:
> All these null checks are kind of useless as the activity should abort onCreate if no track is there, right?

I think so. Same for trackstatistics, right?

In recorded fragment track and trackstatistics cannot be null.

I'm going to open an Issue and look at it carefully.